pandas/core/groupby.py

```diff
@@ -1219,6 +1219,53 @@ class GroupBy(_GroupBy):
     """
     _apply_whitelist = _common_apply_whitelist
 
+    def _bool_agg(self, how, skipna):
+        """Shared func to call any / all Cython GroupBy implementations"""
+
+        def objs_to_bool(vals):
+            try:
+                vals = vals.astype(np.bool)
+            except ValueError:  # for objects
+                vals = np.array([bool(x) for x in vals])
```

Comment:
do we have an asv that hits this (the object path)

Reply:
We do not but happy to add. As far as objects are concerned, I see we have existing tests for `first`, `last`, `nth`, and `count`. I believe the gap there to be `any`, `all`, `bfill`, `cumcount`, `ffill`, `head`, `nunique`, `shift`, `size`, `tail`, `unique` and `value_counts`.

I'd suggest we group all of these into a `GroupByMethodsObjects` class (similar to the one for int / float). If you agree, would you want to move `first`, `last`, `nth` and `count` into that class or keep where they currently are? Moving them would keep the code-base more concise, but I'm not clear on how that would impact the history of benchmarks that you keep

Reply:
ok to move
more important that the code is organized